pkg/theia/commands/utils.go

```diff
@@ -49,6 +57,96 @@ func CreateK8sClient(kubeconfig string) (kubernetes.Interface, error) {
 	return clientset, nil
 }
 
+func setupTheiaClientAndConnection(cmd *cobra.Command, useClusterIP bool) (restclient.Interface, *portforwarder.PortForwarder, error) {
+	kubeconfig, err := ResolveKubeConfig(cmd)
+	if err != nil {
+		return nil, nil, fmt.Errorf("couldn't resolve kubeconfig: %v", err)
+	}
+	clientset, err := CreateK8sClient(kubeconfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("couldn't create k8s client using given kubeconfig, %v", err)
+	}
+	theiaClient, portForward, err := CreateTheiaManagerClient(clientset, kubeconfig, useClusterIP)
+	if err != nil {
+		return nil, nil, fmt.Errorf("couldn't create theia client: %v", err)
+	}
+	return theiaClient.CoreV1().RESTClient(), portForward, err
+}
+
+func CreateTheiaManagerClient(k8sClient kubernetes.Interface, kubeconfig string, useClusterIP bool) (kubernetes.Interface, *portforwarder.PortForwarder, error) {
+	// check and get ca-cert.pem file
+	caCrt, err := GetCaCrt(k8sClient)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error when getting ca-crt: %v", err)
+	}
+	// check and get token
+	token, err := GetToken(k8sClient)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error when getting token: %v", err)
+	}
+	var host string
+	var portForward *portforwarder.PortForwarder
+	if useClusterIP {
+		serviceIP, servicePort, err := GetServiceAddr(k8sClient, config.TheiaManagerServiceName)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error when getting the Theia Manager Service address: %v", err)
+		}
+		host = net.JoinHostPort(serviceIP, fmt.Sprint(servicePort))
+	} else {
+		listenAddress := "localhost"
+		listenPort := apis.TheiaManagerAPIPort
+		_, servicePort, err := GetServiceAddr(k8sClient, config.TheiaManagerServiceName)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error when getting the Theia Manager Service port: %v", err)
+		}
+		// Forward the Theia Manager service port
+		portForward, err = StartPortForward(kubeconfig, config.TheiaManagerServiceName, servicePort, listenAddress, listenPort)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error when forwarding port: %v", err)
+		}
+		host = net.JoinHostPort(listenAddress, fmt.Sprint(listenPort))
+	}
+
+	clientConfig := &restclient.Config{
+		Host:        host,
+		BearerToken: token,
+		TLSClientConfig: restclient.TLSClientConfig{
+			Insecure:   false,
```

Comment:
do we want to expose `Insecure` as cli arg to be fail-safe?

Reply:
Maybe not? If we need Insecure=true, does it mean we have a wrong ca-crt? Or maybe we have some issues in Theia-manager? For both cases, do we still want to connect to Theia-manager?